src/test/java/br/unicamp/meca/mind/action/Test1ActivityCodelet.java

```diff
@@ -26,6 +26,12 @@ public Test1ActivityCodelet(String id, ArrayList<String> perceptualCodeletsIds,
 			ArrayList<String> motivationalCodeletsIds, String motorCodeletId, String soarCodeletId) {
 		super(id, perceptualCodeletsIds, motivationalCodeletsIds, motorCodeletId, soarCodeletId);
 	}
+        
+        @Override
+	public void doConclusion(ArrayList<Memory> perceptualMemories, Memory broadcastMemory, Memory motorMemory) {
```

Comment:
If the objective of the PR changes was having a way to clean the motorMemory container after an execution to avoid problems, why haven't you implemented one test that emulates this very problem and shown how this new mechanism is able to avoid problems? I was expecting to see one of this `doConclusion` to be cleaning motorMemory and the test asserting something in this direction of avoiding the problem you described.

Reply:
In some sense, I created that test with the TestActivityTrackingCodelet. But I just "simulated" that by manipulating the flags, in the same way that doConclusion does. This is because, in order to have a proper test for the mechanism, I would require an Integration Test, and not a Unit Test, because this would require at least two different ActivityCodelets (the current, executing proc, and the last, executing doConclusion), integrated with the proper ActionSteps, and an ActionSequencePlan. But I will try to make a second test within TestActivityTrackingCodelet in order do use the doConclusion, as you propose.

Reply:
I created a second test at TestActivityTrackingCodelet using the doConclusion and showing the expected behavior.